<a href="https://colab.research.google.com/github/carlosfmorenog/CM4125/blob/main/CM4125_T5/CM4125_Lab5_solved.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# CM4125 Topic 5 Lab (Solved)

For this lab, we will be importing the data set `films.csv`

In [ ]:
import requests
import zipfile
import io
import pandas as pd

url = 'https://www.dropbox.com/scl/fo/mh1mnvgmsh8h7819bq1re/AId7Kit6J3BJXTCYZSnQJ78?rlkey=t2tf8yw0onroyv6mxrlm4ipdp&raw=1'
response = requests.get(url)

# Check if the request was successful
response.raise_for_status()

# Open the zip file from the response content
with zipfile.ZipFile(io.BytesIO(response.content)) as z:
    # Extract films.csv
    with z.open('films.csv') as f:
        # Read the extracted CSV into a pandas DataFrame, specifying the encoding and index column
        films = pd.read_csv(f, index_col=0, encoding='latin-1')

films

rank_afc_2007  rank_afc_1998  rank_wga_2005  rank_ss_2012  \
imdb_id                                                                
tt0000417            NaN            NaN            NaN           NaN   
tt0000439            NaN            NaN            NaN           NaN   
tt0006206            NaN            NaN            NaN           NaN   
tt0004972            NaN           44.0            NaN           NaN   
tt0006864           49.0            NaN            NaN           NaN   
...                  ...            ...            ...           ...   
tt5895028            NaN            NaN            NaN           NaN   
tt5168192            NaN            NaN            NaN           NaN   
tt2543164            NaN            NaN            NaN           NaN   
tt2582782            NaN            NaN            NaN           NaN   
tt5663482            NaN            NaN            NaN           NaN   

          included_in_guardian_list included_in_1001_book  \
imdb_id                                                     
tt0000417                       NaN                  True   
tt0000439                       NaN                  True   
tt0006206                      True                  True   
tt0004972                       NaN                  True   
tt0006864                       NaN                  True   
...                             ...                   ...   
tt5895028                       NaN                  True   
tt5168192                       NaN                  True   
tt2543164                       NaN                  True   
tt2582782                       NaN                  True   
tt5663482                       NaN                  True   

                                                      title  year      rated  \
imdb_id                                                                        
tt0000417                                A Trip to the Moon  1902       TV-G   
tt0000439                           The Great Train Robbery  1903       TV-G   
tt0006206                                      Les vampires  1915  NOT RATED   
tt0004972                             The Birth of a Nation  1915  NOT RATED   
tt0006864  Intolerance: Love's Struggle Throughout the Ages  1916  NOT RATED   
...                                                     ...   ...        ...   
tt5895028                                              13th  2016      TV-MA   
tt5168192                                   I, Daniel Blake  2016          R   
tt2543164                                           Arrival  2016      PG-13   
tt2582782                                Hell or High Water  2016          R   
tt5663482                                       Jungle Book  2016         PG   

            released  runtime                        genre  \
imdb_id                                                      
tt0000417   4-Oct-02       13    Short, Adventure, Fantasy   
tt0000439   1-Dec-03       11         Short, Action, Crime   
tt0006206  23-Nov-16      421     Action, Adventure, Crime   
tt0004972   3-Mar-15      165          Drama, History, War   
tt0006864   5-Sep-16      197          Drama, History, War   
...              ...      ...                          ...   
tt5895028   7-Oct-16      100  Documentary, Crime, History   
tt5168192   9-Jun-17      100                        Drama   
tt2543164  11-Nov-16      116       Drama, Mystery, Sci-Fi   
tt2582782  26-Aug-16      102       Crime, Drama, Thriller   
tt5663482  15-Apr-16      106     Adventure, Drama, Family   

                     director  \
imdb_id                         
tt0000417  Georges M_©li_¨s   
tt0000439     Edwin S. Porter   
tt0006206     Louis Feuillade   
tt0004972       D.W. Griffith   
tt0006864       D.W. Griffith   
...                       ...   
tt5895028        Ava DuVernay   
tt5168192           Ken Loach   
tt2543164    Denis Villeneuve   
tt2582782     David Mackenzie   
tt5663482         Jon Favreau   

  

Use Pandas' `read_csv` function, specifying column $0$ as the index column. This column is the `imdb_id`, a unique identifier assigned by the Internet Movie Database (IMDB)

Store the dataframe into an appropriately named value, such as `films` or another name of your choice

Use `head` to show the first $3$ rows

In [ ]:
films.head(3)

rank_afc_2007  rank_afc_1998  rank_wga_2005  rank_ss_2012  \
imdb_id                                                                
tt0000417            NaN            NaN            NaN           NaN   
tt0000439            NaN            NaN            NaN           NaN   
tt0006206            NaN            NaN            NaN           NaN   

          included_in_guardian_list included_in_1001_book  \
imdb_id                                                     
tt0000417                       NaN                  True   
tt0000439                       NaN                  True   
tt0006206                      True                  True   

                             title  year      rated   released  runtime  \
imdb_id                                                                   
tt0000417       A Trip to the Moon  1902       TV-G   4-Oct-02       13   
tt0000439  The Great Train Robbery  1903       TV-G   1-Dec-03       11   
tt0006206             Les vampires  1915  NOT RATED  23-Nov-16      421   

                               genre            director           writer  \
imdb_id                                                                     
tt0000417  Short, Adventure, Fantasy  Georges M_©li_¨s              NaN   
tt0000439       Short, Action, Crime     Edwin S. Porter              NaN   
tt0006206   Action, Adventure, Crime     Louis Feuillade  Louis Feuillade   

          language country         awards  metascore  imdb_rating  imdb_votes  
imdb_id                                                                        
tt0000417      NaN  France            NaN        NaN          8.2       28826  
tt0000439  English     USA         1 win.        NaN          7.3       11609  
tt0006206   French  France  1 nomination.        NaN          6.8        3139

You should see that the index of these rows is `tt0000417`, `tt0000439`, and `tt0006206`

In this data set we will only be using a few columns:

- `imdb_id` is the unique index of the film in the *Internet Movie DataBase*
- `title` is the title of the film
- `year` is the year the film was released
- `imdb_votes` is the number of rating the film has received from users.

Check that you get the same result by using `iloc` with a numerical range `0:3`

In [ ]:
films.iloc[0:3]

rank_afc_2007  rank_afc_1998  rank_wga_2005  rank_ss_2012  \
imdb_id                                                                
tt0000417            NaN            NaN            NaN           NaN   
tt0000439            NaN            NaN            NaN           NaN   
tt0006206            NaN            NaN            NaN           NaN   

          included_in_guardian_list included_in_1001_book  \
imdb_id                                                     
tt0000417                       NaN                  True   
tt0000439                       NaN                  True   
tt0006206                      True                  True   

                             title  year      rated   released  runtime  \
imdb_id                                                                   
tt0000417       A Trip to the Moon  1902       TV-G   4-Oct-02       13   
tt0000439  The Great Train Robbery  1903       TV-G   1-Dec-03       11   
tt0006206             Les vampires  1915  NOT RATED  23-Nov-16      421   

                               genre            director           writer  \
imdb_id                                                                     
tt0000417  Short, Adventure, Fantasy  Georges M_©li_¨s              NaN   
tt0000439       Short, Action, Crime     Edwin S. Porter              NaN   
tt0006206   Action, Adventure, Crime     Louis Feuillade  Louis Feuillade   

          language country         awards  metascore  imdb_rating  imdb_votes  
imdb_id                                                                        
tt0000417      NaN  France            NaN        NaN          8.2       28826  
tt0000439  English     USA         1 win.        NaN          7.3       11609  
tt0006206   French  France  1 nomination.        NaN          6.8        3139

Use 'loc' with a range to retrieve the same three rows

In [ ]:
films.loc['tt0000417':'tt0006206']

rank_afc_2007  rank_afc_1998  rank_wga_2005  rank_ss_2012  \
imdb_id                                                                
tt0000417            NaN            NaN            NaN           NaN   
tt0000439            NaN            NaN            NaN           NaN   
tt0006206            NaN            NaN            NaN           NaN   

          included_in_guardian_list included_in_1001_book  \
imdb_id                                                     
tt0000417                       NaN                  True   
tt0000439                       NaN                  True   
tt0006206                      True                  True   

                             title  year      rated   released  runtime  \
imdb_id                                                                   
tt0000417       A Trip to the Moon  1902       TV-G   4-Oct-02       13   
tt0000439  The Great Train Robbery  1903       TV-G   1-Dec-03       11   
tt0006206             Les vampires  1915  NOT RATED  23-Nov-16      421   

                               genre            director           writer  \
imdb_id                                                                     
tt0000417  Short, Adventure, Fantasy  Georges M_©li_¨s              NaN   
tt0000439       Short, Action, Crime     Edwin S. Porter              NaN   
tt0006206   Action, Adventure, Crime     Louis Feuillade  Louis Feuillade   

          language country         awards  metascore  imdb_rating  imdb_votes  
imdb_id                                                                        
tt0000417      NaN  France            NaN        NaN          8.2       28826  
tt0000439  English     USA         1 win.        NaN          7.3       11609  
tt0006206   French  France  1 nomination.        NaN          6.8        3139

Get a random sampling of $5$ films to display in the notebook

In [ ]:
films.sample(5)

rank_afc_2007  rank_afc_1998  rank_wga_2005  rank_ss_2012  \
imdb_id                                                                
tt0060138            NaN            NaN            NaN          16.0   
tt0211915            NaN            NaN            NaN           NaN   
tt0035169            NaN            NaN            NaN           NaN   
tt0006206            NaN            NaN            NaN           NaN   
tt0086879            NaN           53.0           73.0           NaN   

          included_in_guardian_list included_in_1001_book  \
imdb_id                                                     
tt0060138                       NaN                  True   
tt0211915                       NaN                  True   
tt0035169                       NaN                  True   
tt0006206                      True                  True   
tt0086879                       NaN                  True   

                          title  year      rated   released  runtime  \
imdb_id                                                                
tt0060138   Au Hasard Balthazar  1966  NOT RATED  25-May-66       95   
tt0211915               Amélie  2001          R   8-Feb-02      122   
tt0035169  The Palm Beach Story  1942     PASSED   7-Nov-42       88   
tt0006206          Les vampires  1915  NOT RATED  23-Nov-16      421   
tt0086879               Amadeus  1984          R   5-Apr-85      160   

                               genre            director  \
imdb_id                                                    
tt0060138                      Drama      Robert Bresson   
tt0211915            Comedy, Romance  Jean-Pierre Jeunet   
tt0035169            Romance, Comedy     Preston Sturges   
tt0006206   Action, Adventure, Crime     Louis Feuillade   
tt0086879  Biography, Drama, History        Milos Forman   

                                                      writer  \
imdb_id                                                        
tt0060138                                     Robert Bresson   
tt0211915  Guillaume Laurant (scenario), Jean-Pierre Jeun...   
tt0035169                                    Preston Sturges   
tt0006206                                    Louis Feuillade   
tt0086879  Peter Shaffer (original stage play), Peter Sha...   

                                  language          country  \
imdb_id                                                       
tt0060138                    French, Latin   France, Sweden   
tt0211915                  French, Russian  France, Germany   
tt0035169                          English              USA   
tt0006206                           French           France   
tt0086879  English, Italian, Latin, German      USA, France   

                                                      awards  metascore  \
imdb_id                                                                   
tt0060138                             5 wins & 1 nomination.      100.0   
tt0211915  Nominated for 5 Oscars. Another 56 wins & 61 n...       69.0   
tt0035169                                                NaN        NaN   
tt0006206                                      1 nomination.        NaN   
tt0086879    Won 8 Oscars. Another 33 wins & 14 nominations.       93.0   

           imdb_rating  imdb_votes  
imdb_id                             
tt0060138          7.9       11841  
tt0211915          8.4      540082  
tt0035169          7.7        8470  
tt0006206          6.8        3139  
tt0086879          8.3      272663

Re-run the cell a few times to check that it is randomly choosing each time

**Q. Which film has the index `tt0435761`?**

In [ ]:
films.loc['tt0435761']

rank_afc_2007                                                              NaN
rank_afc_1998                                                              NaN
rank_wga_2005                                                              NaN
rank_ss_2012                                                               NaN
included_in_guardian_list                                                  NaN
included_in_1001_book                                                     True
title                                                              Toy Story 3
year                                                                      2010
rated                                                                        G
released                                                             18-Jun-10
runtime                                                                    103
genre                                             Animation, Adventure, Comedy
director                                                           Lee Unkrich
writer                       John Lasseter (story by), Andrew Stanton (stor...
language                                                      English, Spanish
country                                                                    USA
awards                         Won 2 Oscars. Another 58 wins & 89 nominations.
metascore                                                                 92.0
imdb_rating                                                                8.3
imdb_votes                                                              549330
Name: tt0435761, dtype: object

Answer: Toy Story 3

Display a `DataFrame` containing only films `tt0015361` and `tt0059817`

In [ ]:
films.loc[['tt0015361', 'tt0059817']]

rank_afc_2007  rank_afc_1998  rank_wga_2005  rank_ss_2012  \
imdb_id                                                                
tt0015361            NaN            NaN            NaN           NaN   
tt0059817            NaN            NaN            NaN           NaN   

          included_in_guardian_list included_in_1001_book           title  \
imdb_id                                                                     
tt0015361                       NaN                  True          Strike   
tt0059817                       NaN                  True  Tokyo Olympiad   

           year      rated   released  runtime               genre  \
imdb_id                                                              
tt0015361  1925  NOT RATED        NaN       82               Drama   
tt0059817  1965  NOT RATED  28-Jul-65       93  Documentary, Sport   

                       director  \
imdb_id                           
tt0015361  Sergei M. Eisenstein   
tt0059817          Kon Ichikawa   

                                                      writer  language  \
imdb_id                                                                  
tt0015361  Grigori Aleksandrov, Sergei M. Eisenstein, Ily...       NaN   
tt0059817  Kon Ichikawa, Yoshio Shirasaka, Shuntar_´ Tan...  Japanese   

                country                   awards  metascore  imdb_rating  \
imdb_id                                                                    
tt0015361  Soviet Union                      NaN        NaN          7.6   
tt0059817         Japan  4 wins & 2 nominations.        NaN          8.0   

           imdb_votes  
imdb_id                
tt0015361        4904  
tt0059817        1094

**Q. Which two films are these?**

Answer: Strike and Tokyo Olympiad

Using `iloc`, display a dataframe containing only rows indexed $57$, $79$, and $190$

In [ ]:
films.iloc[[57, 79, 190]]

rank_afc_2007  rank_afc_1998  rank_wga_2005  rank_ss_2012  \
imdb_id                                                                
tt0022100            NaN            NaN            NaN           NaN   
tt0024028            NaN            NaN            NaN           NaN   
tt0038890            NaN            NaN            NaN           NaN   

          included_in_guardian_list included_in_1001_book             title  \
imdb_id                                                                       
tt0022100                       NaN                  True                 M   
tt0024028                       NaN                  True  Footlight Parade   
tt0038890                      True                  True   Rome, Open City   

           year      rated   released  runtime                     genre  \
imdb_id                                                                    
tt0022100  1931  NOT RATED  31-Aug-31       99     Crime, Drama, Mystery   
tt0024028  1933          G  21-Oct-33      104  Comedy, Musical, Romance   
tt0038890  1945   APPROVED  27-Sep-45      103      Drama, Thriller, War   

                     director  \
imdb_id                         
tt0022100          Fritz Lang   
tt0024028         Lloyd Bacon   
tt0038890  Roberto Rossellini   

                                                      writer  \
imdb_id                                                        
tt0022100      Thea von Harbou (script), Fritz Lang (script)   
tt0024028  Manuel Seff (screen play), James Seymour (scre...   
tt0038890  Sergio Amidei (screenplay), Federico Fellini (...   

                         language  country  \
imdb_id                                      
tt0022100                  German  Germany   
tt0024028         English, French      USA   
tt0038890  German, Italian, Latin    Italy   

                                           awards  metascore  imdb_rating  \
imdb_id                                                                     
tt0022100                                 2 wins.        NaN          8.4   
tt0024028                                  1 win.        NaN          7.8   
tt0038890  Nominated for 1 Oscar. Another 7 wins.        NaN          8.1   

           imdb_votes  
imdb_id                
tt0022100       99611  
tt0024028        3367  
tt0038890       16080

Using `loc`, display the same three films

In [ ]:
films.loc[['tt0022100', 'tt0024028', 'tt0038890']]

rank_afc_2007  rank_afc_1998  rank_wga_2005  rank_ss_2012  \
imdb_id                                                                
tt0022100            NaN            NaN            NaN           NaN   
tt0024028            NaN            NaN            NaN           NaN   
tt0038890            NaN            NaN            NaN           NaN   

          included_in_guardian_list included_in_1001_book             title  \
imdb_id                                                                       
tt0022100                       NaN                  True                 M   
tt0024028                       NaN                  True  Footlight Parade   
tt0038890                      True                  True   Rome, Open City   

           year      rated   released  runtime                     genre  \
imdb_id                                                                    
tt0022100  1931  NOT RATED  31-Aug-31       99     Crime, Drama, Mystery   
tt0024028  1933          G  21-Oct-33      104  Comedy, Musical, Romance   
tt0038890  1945   APPROVED  27-Sep-45      103      Drama, Thriller, War   

                     director  \
imdb_id                         
tt0022100          Fritz Lang   
tt0024028         Lloyd Bacon   
tt0038890  Roberto Rossellini   

                                                      writer  \
imdb_id                                                        
tt0022100      Thea von Harbou (script), Fritz Lang (script)   
tt0024028  Manuel Seff (screen play), James Seymour (scre...   
tt0038890  Sergio Amidei (screenplay), Federico Fellini (...   

                         language  country  \
imdb_id                                      
tt0022100                  German  Germany   
tt0024028         English, French      USA   
tt0038890  German, Italian, Latin    Italy   

                                           awards  metascore  imdb_rating  \
imdb_id                                                                     
tt0022100                                 2 wins.        NaN          8.4   
tt0024028                                  1 win.        NaN          7.8   
tt0038890  Nominated for 1 Oscar. Another 7 wins.        NaN          8.1   

           imdb_votes  
imdb_id                
tt0022100       99611  
tt0024028        3367  
tt0038890       16080

Create a dataframe containing all films in the data set where the `year` is $2000$ or later

Store this new dataset in a new variable called `films2k`

In [ ]:
films2k = films[films['year'] >= 2000]

**Q. How many films from this data set are from $2000$ or later?**

In [ ]:
len(films2k)

206

Answer: 206

**Q. What is the median (`50%`) `runtime` of the films in the data set from the year $2000$ onwards?**

Hint: you can use the `describe` function to see the statistics of a dataframe

In [ ]:
films2k.describe()

rank_afc_2007  rank_afc_1998  rank_wga_2005  rank_ss_2012         year  \
count            1.0            0.0       3.000000      2.000000   206.000000   
mean            50.0            NaN      89.000000     26.000000  2007.368932   
std              NaN            NaN      11.532563      2.828427     5.003395   
min             50.0            NaN      77.000000     24.000000  2000.000000   
25%             50.0            NaN      83.500000     25.000000  2003.000000   
50%             50.0            NaN      90.000000     26.000000  2007.000000   
75%             50.0            NaN      95.000000     27.000000  2012.000000   
max             50.0            NaN     100.000000     28.000000  2016.000000   

          runtime   metascore  imdb_rating    imdb_votes  
count  206.000000  202.000000   206.000000  2.060000e+02  
mean   121.310680   80.980198     7.720388  2.546437e+05  
std     29.785093    9.509859     0.509013  2.916283e+05  
min     80.000000   47.000000     5.900000  1.200000e+01  
25%    102.000000   74.250000     7.400000  3.857050e+04  
50%    118.000000   82.000000     7.700000  1.525265e+05  
75%    134.000000   88.000000     8.000000  3.637630e+05  
max    383.000000  100.000000     9.000000  1.699835e+06

Answer: 118 minutes

**Q. What is the median `runtime` for films between the years $1990$ and $1999$ in the films data set?**

In [ ]:
films90s = films[(films['year'] >= 1990) & (films['year'] < 2000)]
films90s.describe()

rank_afc_2007  rank_afc_1998  rank_wga_2005  rank_ss_2012         year  \
count      11.000000       8.000000      17.000000      2.000000   165.000000   
mean       75.090909      73.875000      46.470588     38.500000  1994.551515   
std        24.598596      28.866379      20.877972      4.949747     2.905852   
min         8.000000       9.000000      16.000000     35.000000  1990.000000   
25%        71.500000      69.500000      30.000000     36.750000  1992.000000   
50%        76.000000      79.500000      41.000000     38.500000  1995.000000   
75%        90.500000      94.250000      61.000000     40.250000  1997.000000   
max        99.000000      98.000000      89.000000     42.000000  1999.000000   

          runtime   metascore  imdb_rating    imdb_votes  
count  165.000000  117.000000   165.000000  1.650000e+02  
mean   123.139394   76.726496     7.665455  1.921053e+05  
std     40.977968   10.430909     0.573038  3.128316e+05  
min     73.000000   47.000000     6.100000  1.820000e+02  
25%    100.000000   69.000000     7.300000  1.371500e+04  
50%    112.000000   77.000000     7.700000  5.055800e+04  
75%    134.000000   85.000000     8.100000  1.914860e+05  
max    450.000000  100.000000     9.300000  1.711064e+06

Answer: 112 minutes

Create a dataframe of 90's films, sorted by `title`, in the variable `films90s_sorted`.

In [ ]:
films90s_sorted = films90s.sort_values('title')

Display the first $5$ items in `films90s_sorted`

In [ ]:
films90s_sorted.head(5)

rank_afc_2007  rank_afc_1998  rank_wga_2005  rank_ss_2012  \
imdb_id                                                                
tt0101985            NaN            NaN            NaN           NaN   
tt0104008            NaN            NaN            NaN           NaN   
tt0103634            NaN            NaN            NaN           NaN   
tt0099012            NaN            NaN            NaN           NaN   
tt0185125            NaN            NaN            NaN           NaN   

          included_in_guardian_list included_in_1001_book  \
imdb_id                                                     
tt0101985                       NaN                  True   
tt0104008                       NaN                  True   
tt0103634                       NaN                  True   
tt0099012                       NaN                  True   
tt0185125                       NaN                  True   

                                                    title  year      rated  \
imdb_id                                                                      
tt0101985                           A Brighter Summer Day  1991  NOT RATED   
tt0104008                                A Tale of Winter  1992        NaN   
tt0103634  Aileen Wuornos: The Selling of a Serial Killer  1992          R   
tt0099012                                           Alice  1990      PG-13   
tt0185125                             All About My Mother  1999          R   

            released  runtime                     genre          director  \
imdb_id                                                                     
tt0101985  25-Nov-11      237     Crime, Drama, Romance       Edward Yang   
tt0104008  29-Jan-92      114                     Drama     _‰ric Rohmer   
tt0103634   4-Feb-94       87        Documentary, Crime   Nick Broomfield   
tt0099012  10-Jan-91      102  Comedy, Fantasy, Romance       Woody Allen   
tt0185125  31-Mar-00      101                     Drama  Pedro Almod__var   

                                                    writer  \
imdb_id                                                      
tt0101985  Hung Hung, Mingtang Lai, Alex Yang, Edward Yang   
tt0104008                                    _‰ric Rohmer   
tt0103634                                  Nick Broomfield   
tt0099012                                      Woody Allen   
tt0185125                                 Pedro Almod__var   

                                  language        country  \
imdb_id                                                     
tt0101985  Mandarin, Min Nan, Shanghainese         Taiwan   
tt0104008                           French         France   
tt0103634                          English             UK   
tt0099012               English, Cantonese            USA   
tt0185125                 Spanish, Catalan  Spain, France   

                                                      awards  metascore  \
imdb_id                                                                   
tt0101985                           8 wins & 12 nominations.       90.0   
tt0104008                             2 wins & 1 nomination.        NaN   
tt0103634                                      1 nomination.        NaN   
tt0099012  Nominated for 1 Oscar. Another 1 win & 5 nomin...        NaN   
tt0185125     Won 1 Oscar. Another 56 wins & 37 nominations.       87.0   

           imdb_rating  imdb_votes  
imdb_id                             
tt0101985          8.4        2872  
tt0104008          7.6        2574  
tt0103634          7.0        1618  
tt0099012          6.6       10644  
tt0185125          7.9       69368

Show the 90's films sorted in **descending** order by `imbd_votes`, displaying only the first $10$

In [ ]:
films90s_by_votes = films90s.sort_values('imdb_votes', ascending=False)
films90s_by_votes.head(10)

rank_afc_2007  rank_afc_1998  rank_wga_2005  rank_ss_2012  \
imdb_id                                                                
tt0111161           72.0            NaN           22.0           NaN   
tt0137523            NaN            NaN            NaN           NaN   
tt0110912           94.0           95.0           16.0           NaN   
tt0109830           76.0           71.0           89.0           NaN   
tt0133093            NaN            NaN            NaN           NaN   
tt0114369            NaN            NaN            NaN           NaN   
tt0102926           74.0           65.0           61.0           NaN   
tt0120815           71.0            NaN            NaN           NaN   
tt0108052            8.0            9.0           49.0           NaN   
tt0169547            NaN            NaN           38.0           NaN   

          included_in_guardian_list included_in_1001_book  \
imdb_id                                                     
tt0111161                       NaN                  True   
tt0137523                       NaN                  True   
tt0110912                      True                  True   
tt0109830                       NaN                  True   
tt0133093                      True                  True   
tt0114369                       NaN                  True   
tt0102926                       NaN                  True   
tt0120815                       NaN                  True   
tt0108052                       NaN                  True   
tt0169547                       NaN                  True   

                              title  year  rated   released  runtime  \
imdb_id                                                                
tt0111161  The Shawshank Redemption  1994      R  14-Oct-94      142   
tt0137523                Fight Club  1999      R  15-Oct-99      139   
tt0110912              Pulp Fiction  1994      R  14-Oct-94      154   
tt0109830              Forrest Gump  1994  PG-13   6-Jul-94      142   
tt0133093                The Matrix  1999      R  31-Mar-99      136   
tt0114369                     Se7en  1995      R  22-Sep-95      127   
tt0102926  The Silence of the Lambs  1991      R  14-Feb-91      118   
tt0120815       Saving Private Ryan  1998      R  24-Jul-98      169   
tt0108052          Schindler's List  1993      R   4-Feb-94      195   
tt0169547           American Beauty  1999      R   1-Oct-99      122   

                               genre                         director  \
imdb_id                                                                 
tt0111161               Crime, Drama                   Frank Darabont   
tt0137523                      Drama                    David Fincher   
tt0110912               Crime, Drama                Quentin Tarantino   
tt0109830              Comedy, Drama                  Robert Zemeckis   
tt0133093             Action, Sci-Fi  Lana Wachowski, Lilly Wachowski   
tt0114369      Crime, Drama, Mystery                    David Fincher   
tt0102926     Crime, Drama, Thriller                   Jonathan Demme   
tt0120815         Action, Drama, War                 Steven Spielberg   
tt0108052  Biography, Drama, History                 Steven Spielberg   
tt0169547             Drama, Romance                       Sam Mendes   

                                                      writer  \
imdb_id                                                        
tt0111161  Stephen King (short story "Rita Hayworth and S...   
tt0137523     Chuck Palahniuk (novel), Jim Uhls (screenplay)   
tt0110912  Quentin Tarantino (story), Roger Avary (story)...   
tt0109830      Winston Groom (novel), Eric Roth (screenplay)   
tt0133093                    Lilly Wachowski, Lana Wachowski   
tt0114369                                Andrew Kevin Walker   
tt0102926      Thomas Harris (novel), Ted Tally (screenplay)   
tt0120815                                       Robert Rodat   
tt0108052  Thomas Keneally (boo

You should see that the most voted film (The Shawshank Redemption) has $1711064$ votes

**Q. Which 90's film received the second-most votes?**

In [ ]:
films90s_by_votes.iloc[1]['title']

'Fight Club'

Answer: Fight Club

**Q. Which 90's film received the twentieth-most votes?**

In [ ]:
films90s_by_votes.iloc[19]['title']

'Jurassic Park'

Answer: Jurassic Park